### PR TITLE
Rename errors

### PR DESCRIPTION
**Pull request by Wondertan**
Part of libp2p/go-libp2p#814

### CODE REVIEW

File: const.go [hunk 1]
```diff
@@ -5,69 +5,69 @@ import (
 	"fmt"
 )
 
-type YamuxError struct {
+type Error struct {
```
Comment (Wondertan) [inline const.go]: Previously was started with repository suffix.

File: const.go [hunk 2]
```diff
@@ -5,69 +5,69 @@ import (
...
 
 	// ErrRemoteGoAway is used when we get a go away from the other side
-	ErrRemoteGoAway = &YamuxError{msg: "remote end is not accepting connections"}
+	ErrRemoteGoAway = &Error{msg: "remote end is not accepting connections"}
 
-	// ErrConnectionReset is sent if a stream is reset. This can happen
+	// ErrStreamReset is sent if a stream is reset. This can happen
 	// if the backlog is exceeded, or if there was a remote GoAway.
-	ErrConnectionReset = &YamuxError{msg: "stream reset"}
+	ErrStreamReset = &Error{msg: "stream reset"}
```
Comment (Wondertan) [inline const.go]: Not a connection reset, but a stream one.

**Comment by Wondertan** [comment]
@Stebalien, what about merging this first and then pointing `go-libp2p-yamux` to merged updates here?

**Comment by Stebalien** [comment]
Done and released.

**Pull request merged**